lib/src/object_id.rs

```diff
@@ -103,14 +103,19 @@ macro_rules! impl_id_type {
 
 pub(crate) use {id_type, impl_id_type};
 
+/// Stores bytes representing the minimum length unique hexadecimal prefix of an
+/// identifier.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct HexPrefix {
-    // For odd-length prefix, lower 4 bits of the last byte is padded with 0
+    // For odd-length prefixes, the lower 4 bits of the last byte are
+    // zero-filled (e.g. the prefix "abc" is stored in two bytes as 0xabc0).
     min_prefix_bytes: Vec<u8>,
     has_odd_byte: bool,
 }
 
 impl HexPrefix {
+    /// Decodes `prefix` to bytes and returns a new `HexPrefix`. Returns `None`
```

Comment:
nit: I feel like the "Decodes `prefix` to bytes" bit is an implementation detail that's not important to the caller.

Reply:
I reworded this slightly. I think the caller just needs to know that `None` is returned if "prefix" can't be decoded from hex to bytes.